src/main/java/net/ossindex/gradle/OssIndexPlugin.java

```diff
@@ -22,20 +23,41 @@
 
 public class OssIndexPlugin implements Plugin<Project> {
 
+    public static JunitXmlReportWriter junitXmlReportWriter = null;
+
+    public static String junitReport = null;
+    public static Integer instanceId;
+
+    private static AuditExtensions settings = null;
+
     private static final Logger logger = LoggerFactory.getLogger(OssIndexPlugin.class);
     private List<Proxy> proxies = new LinkedList<>();
 
     private AuditorFactory factory = new AuditorFactory();
+    private AuditResultReporter reporter = null;
 
     public void setAuditorFactory(AuditorFactory factory) {
         this.factory = factory;
     }
+    private Project project = null;
+
+    public OssIndexPlugin() {

```

Comment:
I don't know what the chances are for this ever happening, but technically there is a chance for a race condition on he assignment of the values below. If this constructor is called simultaneously twice, then in *theory* the variable will be null in both cases and can therefore be instantiated twice.

This would be fabulously rare, and likely not possible in this usage as it is a gradle plugin that is likely instantiated in a much more serial way.

To be sure, however, it would probably be best to move the initialization of the variables right up to their declarations, like so:
```
public static JunitXmlReportWriter junitXmlReportWriter =  new JunitXmlReportWriter();

public static String junitReport = null;
public static Integer instanceId = 0; // Technically I think 0 is the default, but I don't mind being explicit
```

Reply:
Hi Ken,

No that wouldn't work. I had to spend a fair bit of time exploring this behaviour because gradle was somewhat less than deterministic. Hopefully just because it was running on my machine or because the project I am testing has grown to a size where it needs rearranging.

Gradle called the constructor 67 times on start of the audit – apart from now and then when it called it twice as often......

Only after that did it cycle through the 67 instances – even if it had created twice as many – calling the apply method. So I couldn't increment in the constructor either, as you would expect, otherwise I ended up never reducing to 0, because I only counted down to 67. 

As all of the instances update the same static report writer, I needed to ensure we only write it out when we're exiting the very last plugin instance.

If I set the instance ID to zero when the plugin is instantiated as you suggest above, then it will always be reset to 0, being set to 1 in the apply method and no higher. So we wouldn't know when we are finishing up in the last instance.

So after all that I had to increment in apply and decrement in doAudit in order to get this to work.

Hope that makes sense.